# Comparison of slideshow creating methods

In [7]:
from usefull_functions import (total_score, 
     create_vertical_slides, 
     load_photos_from_json)
from vertical_photos_combining_methods import  (random_pair_vertical_photos, 
     similar_pair_vertical_photos, 
     different_pair_vertical_photos)
from slideshow_creating_methods import (random_slideshow,
                                        nearest_neighbor_slideshow,
                                        grouped_slideshow,
                                        mixed_slideshow)

## Random method

### Random Vertical photos

In [3]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

174418


### Similar Vertical photos

In [4]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

140996


### Different Vertical photos

In [6]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

185810


## Nearest neighbor (greedy) method

### Random Vertical photos

In [34]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

342713


### Similar Vertical photos

In [35]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

302162


### Different Vertical photos

In [36]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

367586


## Grouping by tags

### Random Vertical photos

In [7]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

222492


### Similar Vertical photos

In [8]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

190445


### Different Vertical photos

In [9]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

236592


## Mixed method

### Random Vertical photos

In [2]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

356534


### Similar Vertical photos

In [3]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

318150


### Different Vertical photos

In [9]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

371991


## Compare afficiency of all methods

|           | Random | Nearest neighbor | Grouped by tags | Mixed  |
|-----------|--------|------------------|-----------------|--------|
| Random    | 174418 | 342713           | 222492          | 356534 |
| Similar   | 140996 | 302162           | 190445          | 318150 |
| Different | 185810 | 367586           | 236592          | 381416 |

Max result - 415180